clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -0,0 +1,366 @@
+    this.store = store;
+    this.params = params;
+    this.expectCmd = generateExpectCmd();
+  }
+
+  public void setParams(Params params) {
+    this.params = params;
+  }
+
+  public String getStore() {
+    return store;
+  }
+
+  public Result getResult() {
+    return result;
+  }
+
+  private List<String> generateRepushCommand() {
+    List<String> cmd = new ArrayList<>();
+    cmd.add(this.params.command);
+    cmd.add(REPUSH_MODULE);
+    cmd.add(REPUSH_SOURCE_KAFKA);
+
+    if (store != null) {
+      cmd.add("--store " + store);
+    }
+
+    if (params.fabric != null) {
+      cmd.add("--fabric " + params.fabric);
+    }
+
+    if (params.rewindTimeOverrideSeconds != -1) {
+      cmd.add("--rewind_time_override_seconds " + params.rewindTimeOverrideSeconds);
+    }
+
+    if (params.fabricGroup != null) {
+      cmd.add("--fabric_group " + params.fabricGroup);
+    }
+
+    if (params.force) {
+      cmd.add("--force");
+    }
+
+    if (params.loginUser != null) {
+      cmd.add("--login_user " + params.loginUser);
+    }
+
+    if (params.proxyUser != null) {
+      cmd.add("--proxy_user " + params.proxyUser);
+    }
+
+    if (params.storelistFile != null) {
+      cmd.add("--storelist_file " + params.storelistFile);
+    }
+
+    if (params.jobConcurrency != -1) {
+      cmd.add("--job_concurrency " + params.jobConcurrency);
+    }
+    return cmd;
+  }
+
+  private List<String> generateExpectCmd() {
+    List<String> expectCmd = new ArrayList<>();
+    // Start a shell process so that it contains the right PATH variables.
+    expectCmd.add("sh");
+    expectCmd.add("-c");
+    /*
+     * repush.py uses getpass.getpass to prompt the user for a password without echoing. It adds difficulties in handling
+     * its input/ouput as by defaults it always reads and writes directly on the tty device i.e. its controlling terminal
+     * (/dev/tty) thus bypassing the stdin, stdout, or stderr.
+     *
+     * We use expect tool to solve this issue because expect tool uses pseudo-terminals (ptys). Ptys are logical device
+     * drivers that gives the program an illusion that it is a real terminal drivers. Programs that open /dev/tty will
+     * actually end up speaking to their pty.
+     *
+     * The following commands:
+     * 1. creates a new process running repush command.
+     * 2. waits until the output of the process to match a pattern (Password + VIP:) from the terminal then send the
+     *    pre-defined password to the process.
+     * 3. exits the process if no such pattern is discovered from the terminal.
+     * 4. waits until the successful or failed messages from Azkaban.
+     */
+    expectCmd.add(
+        "expect -c \"spawn -noecho " + String.join(" ", generateRepushCommand()) + "\n" + "expect {\n"
+            + "{*Password + VIP:*} {send " + this.params.getPassword() + "\\r}\n"
+            + "-re \"^\\(\\(?!Password\\).\\)+\\$\" {exit}\n" + "}\n" + "expect {*https*} {}\n" + "\"");
```

Comment:
This is also a leaky abstraction - it expects the tool to print `Password + VIP` on the screen

Reply:
To your point, I think we can get rid of the usage of `expect` completely if we can make `venice-tool` to take credentials as input directly instead of prompting and asking user to type inputs. However, this requires changes on the `venice-tool` side.